src/main/java/com/samoilenko/onlinebookstore/controller/BookController.java

```diff
@@ -32,32 +33,37 @@ public class BookController {
             summary = "Get all books",
             description = "Returns a list of available books that can be in one page."
     )
+    @PreAuthorize("hasAnyRole('ROLE_USER', 'ROLE_ADMIN')")
     @GetMapping
     public List<BookDto> getAll(Pageable pageable) {
         return bookService.findAll(pageable);
     }
 
     @Operation(summary = "Get book by ID", description = "Returns a single book by its unique ID.")
+    @PreAuthorize("hasAnyRole('ROLE_USER', 'ROLE_ADMIN')")
```

Comment:
we need to add `ROLE_` when we use `hasAuthorithy()` but here we already specified that we are checking exactly role.
Same for other similar cases
```suggestion
    @PreAuthorize("hasAnyRole('USER', 'ADMIN')")
```